pyclient/deephaven/batch_assembler.py

```diff
@@ -0,0 +1,53 @@
+from deephaven.proto import table_pb2
+from deephaven.qst import *
+
+
+class BatchOpAssembler:
+    def __init__(self, session=None):
+        self.session = session
+        self.ops = []
+        self._current_table_ref = None
+
+    @property
+    def batch(self):
+        return self.ops
+
+    def _make_result_id(self, op):
+        if op.child:
+            return self.session.make_flight_ticket()
+        return None
+
+    def _set_last_op(self, op):
+        self._last_op = op
+
+    def visit(self, op):
+        if isinstance(op, NoneOp):
+            self._current_table_ref = table_pb2.TableReference(ticket=op.table.ticket)
+        elif isinstance(op, UpdateOp):
+            self.build_update(op)
+        elif isinstance(op, TailOp):
+            self.build_tail()
+        elif isinstance(op, HeadOp):
+            self.build_head()
```

Comment:
In java, the visit method is implemented by each concrete type to make sure that no "isinstance" checks need to be done. I'm not sure exactly what it would look like here, but maybe something like:

```python
def build(self, op):
  op.walk(BuildVisitor())
```